added Document Synchronization integration tests

steveyegge opened the pull request:
This PR introduces a new set of integration tests for document synchronization testing.

This PR is dependent on a [Cody-side PR](https://github.com/sourcegraph/cody/pull/4512), and the tests will not pass until that PR is merged.

## Test plan

This is not ready to merge yet; I'll update this section when it's ready to go.

Commit:
refactored tests to share test file name

pkukielka commented on src/integrationTest/kotlin/com/sourcegraph/cody/util/CodyIntegrationTestFixture.kt:
```diff
@@ -29,13 +27,23 @@ import java.nio.file.Paths
+        this.javaClass.getMethod(methodName)
+            ?: throw IllegalStateException(
+                "No method with name $methodName found in ${this.javaClass.name}")
+    if (method.isAnnotationPresent(TestFile::class.java)) {
```
Maybe we could use the default test file of no annotation is added?

steveyegge replied:
Good call; I'll clean that up.

pkukielka replied:
Looks like you missed this one @steveyegge

steveyegge replied:
Yeah, there really isn't a good choice for a default right now, since the only test file is specific to `DocumentCodeTest`. 

But I went ahead and used that as the default for now, until we get more tests with disk files.

steveyegge replied:
I had to roll this change back because it broke all the new tests, which don't have testResources files on disk.

pkukielka reviewed (approved):
I added one comment, otherwise LGTM

Commit:
more work on doc sync tests

Commit:
more progress on integration test

they still don't pass but they are getting closer

Commit:
got first doc-sync test actually passing, verified

Commit:
Refactored doc-sync code into a new fixture

Commit:
implemented some more tests; not all are passing yet

Commit:
Got all 11 document synchronization tests working

Commit:
added some assertions

Commit:
some minor refactoring of doc sync tests

dominiccooney reviewed (approved):
Looks *good*. Note, one bit of feedback from @pkukielka you said you'll address.

pkukielka commented on src/integrationTest/kotlin/com/sourcegraph/cody/util/CodyIntegrationTextFixture.kt:
```diff
@@ -42,7 +42,8 @@ open class CodyIntegrationTextFixture : BasePlatformTestCase() {
 
   override fun tearDown() {
     try {
-      FixupService.getInstance(myFixture.project).getActiveSession()?.apply {
```
Shouldn't that whole file be removed and that line modified in the `CodyIntegrationTextFixture.kt`?
Then git should correctly show it as rename, and not like now as whole new file added.

steveyegge replied:
Good catch; fixed that.

Commit:
removed old file

Commit:
addressed PR feedback

Commit:
spotless

Commit:
more cleanup on doc-sync tests

renamed testing/requestWorkspaceDocuments -> testing/workspaceDocuments and put in some more asssertions

Commit:
turn on CODY_RECORD_IF_MISSING for integration tests

This prevents it from spewing an error over a graphql call

Commit:
updated recording

Commit:
updated recording

Commit:
better error messaging around agent startup failures

Commit:
spotless

Commit:
fixed a bug in copying the agent binary to a temp location

Commit:
improved error messaging around remote agent

Commit:
re-enabled DocumentCodeTest

Commit:
re-enabled DocumentCodeTest

Commit:
fixed an NPE during integration testing

Commit:
got the DocumentSynchronizationTest working again

DocumentCodeTest is still failing

Commit:
re-enabled DocumentSynchronizationTest

Commit:
put DocumentCodeTests back

Commit:
merged origin/main

Commit:
spotless

Commit:
spotless

Commit:
updated pinned Cody commit

Commit:
updated pinned Cody commit

Commit:
improved test failure message when error lens is displayed

pkukielka committed:
Few changes to make tests work

Commit:
added more error handling when finding repo roots

was failing during tests

pkukielka committed:
Bump cody commit

mkondratek committed:
Do not list the same account twice (#1864)

## Test plan

Add account that exists 
1. Using settings, add the account that already exists in the list

mkondratek committed:
Do not open subscription page when updating invalid pro account token having a pro llm selected before (#1866)

## Test plan
1. Having a pro account
2. switch to a pro llm in the chat
3. remove the token from the web 
4. re-authenticate via github

pkukielka committed:
Add auth telemetry events (#1863)

Closes
https://linear.app/sourcegraph/issue/CODY-988/p1-authorization-and-log-in-log-out

## Test plan

Repeat for dotcom and enterprise user account:

**Scenario 1** 

1. Remove all existing accounts
2. Make sure you are redirected to the login panel
3. Login using one of the available providers
4. Make sure that both `cody.auth.login/clicked` and
`cody.auth.signin.token/clicked` telemetry events were fired

**Scenario 2** 

1. Click on Cody status bar and choose `Manage Accounts`
2. Click to add new account
3. Make sure that  `cody.auth.login/clicked` was fired
4. Fill the account details (server and token) and confirm 
5.  Make sure that  `cody.auth.signin.token/clicked` was fired

mkondratek committed:
Simplify SignInWithSourcegraphPanel.kt (#1865)

No functional changes. Features should work as before.

## Test plan
1. Sign in panel features working

mkondratek committed:
Show the login screen when the token is missing (#1867)

## Test plan
Delete the token from the keychain externally 
You can run a separate instance of IDE and remove the account there.

1. Log in to the account
2. Open another IDE, log in to the same account, apply, then remove the
account, apply
3. Switch back to the first ide
4. Close and reopen the project
5. Login panel visible

mkondratek committed:
Minor code imporvements (#1870)

## Test plan
1. `Sign In with <>` and `Get started` buttons work

pkukielka committed:
Improve action discoverability by adding Cody: prefix in Search Everywhere (#1872)

Closes
https://linear.app/sourcegraph/issue/CODY-2130/improve-how-actions-appear-in-shiftshift-menu

## Changes

I moved action properties like `text` and `description` to `plugin.xml`.
It is just easier to edit and inspect this way.
To be honest, Geminit 1.5 did most of the work:

> Given the following plugin.xml file and properties file, please modify
plugin.xml so each action have additional propertes `text` and
`description` with values from properties file.
> Additionally for every action please add such tag: `<override-text
place="GoToAction" text="Cody: $TEXT"/>` where `$TEXT` is taken from
properties file as well.
> For some actions like: `<action
id="Cody.Accounts.LogInToSourcegraphAction"
class="com.sourcegraph.cody.config.LogInToSourcegraphAction"/>` you may
have to change them to form such as:
> ```
> <action id="Cody.Accounts.LogInToSourcegraphAction"
class="com.sourcegraph.cody.config.LogInToSourcegraphAction" text="Log
In to Sourcegraph">
> <override-text place="GoToAction" text="Cody: Log In to Sourcegraph"/>
> </action>
>```
>
> Please output whole content of the modified plugin.xml file.


## Test plan

1. Hit `Shift shift`
2. Type 'Cody:'


![image](https://github.com/sourcegraph/jetbrains/assets/1519649/5ffd1d07-0088-4cb6-b5c8-3894b281bbe1)

odisseus committed:
Small improvements to the CI workflow (#1851)

pkukielka committed:
Bump cody commit, add startup telemetry (#1873)

1. Bump cody commit to include changes from
https://github.com/sourcegraph/cody/pull/4766
2. Add startup telemetry event.
I realised that only v2 telemetry events sent directly from IJ contains
info about IDE version and we want to have at least one such event for
every user IDE. Install action telemetry event would not be sufficient
because it would not tell us anything about existing installations.

Same as in https://github.com/sourcegraph/cody/pull/4766

abeatrix committed:
README: fix typo (#1875)

Add the missing value for the ollama docs

## Test plan

<!-- All pull requests REQUIRE a test plan:
https://sourcegraph.com/docs/dev/background-information/testing_principles

Why does it matter?

These test plans are there to demonstrate that are following industry
standards which are important or critical for our customers.
They might be read by customers or an auditor. There are meant be simple
and easy to read. Simply explain what you did to ensure
your changes are correct!

Here are a non exhaustive list of test plan examples to help you:

- Making changes on a given feature or component:
- "Covered by existing tests" or "CI" for the shortest possible plan if
there is zero ambiguity
  - "Added new tests"
- "Manually tested" (if non trivial, share some output, logs, or
screenshot)
- Updating docs:
  - "previewed locally"
  - share a screenshot if you want to be thorough
- Updating deps, that would typically fail immediately in CI if
incorrect
  - "CI"
  - "locally tested"
-->

![Screenshot 2024-07-03 at 12 19
49 PM](https://github.com/sourcegraph/jetbrains/assets/68532117/bb29c397-a40e-453f-b8bf-010e169de863)

mkondratek committed:
Manual Stable Release (#1862)

Our current setup produces a lot of stable releases that often are not
unhidden eventually. That makes the QA and awaiting JB approval process
parallel. Based on the recent experience only 1/8 releases goes public.
That is a waste of JB approval team's time and CI time.

Let's change the setup. The nightly version handling does not change but
the default release script does not longer publish the stable version.
Instead the separate workflow can be triggered on the specific tag that
publishes the stable version.

## Test plan
1. push a tag for a nightly release
2. trigger the stable release manually

to be tested once merged

pkukielka committed:
Add proper groups for Cody actions (#1884)

Closes
https://linear.app/sourcegraph/issue/CODY-2095/organize-keymap-settings

## Changes

1. Proper grouping was added for all actions in plugin.xml
2. The same grouping is used for nice visual organisation of keymap
management

## Test plan

Visual verification in `Settings > Keymap` (see screenshots):


![image](https://github.com/sourcegraph/jetbrains/assets/1519649/b5292cfb-5a2f-47ae-9706-512fb2675b8a)

![image](https://github.com/sourcegraph/jetbrains/assets/1519649/5553077a-c21e-4003-b1c8-0c2f279f6467)

pkukielka committed:
Bump cody commit, adjust agent launch params (#1887)

## Test plan

Full QA according to a test plan.
As for launch script args change that is covered by automatic tests.
